components/RadioButtons.tsx

```diff
@@ -33,24 +32,29 @@ const RadioButtons = ({
 
   return (
     <>
-      {title && <h3>{title}</h3>}
       <fieldset
+        aria-labelledby={`${uuid}-legend`}
         className={styles.container}
         disabled={disabled}
-        onChange={
-          (e) =>
-            updateSelectedOption(parseInt((e.target as HTMLInputElement).value))
-          // eslint-disable-next-line react/jsx-curly-newline
-        }
+        onChange={(e) => {
+          const selectedValue = parseInt((e.target as HTMLInputElement).value)
+          updateSelectedOption(selectedValue)
+        }}
       >
+        {title && (
+          <legend id={`${uuid}-legend`}>
+            <h2>{title}</h2>
+          </legend>
+        )}
         {options?.map((option, index) => (
-          <label key={index}>
+          <label htmlFor={title} key={index}>
             <input
+              aria-checked={index === selectedOption}
```

Comment:
Do we need aria-checked if we have `checked`?

Reply:
I think it's a nice-to-have
![image](https://user-images.githubusercontent.com/62163307/232825596-30bc83ed-739e-4250-8746-fc7dda5ff746.png)

Reply:
Looks like it's not necessary: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked

Reply:
Good find